pages/index.tsx

```diff
@@ -1,9 +1,30 @@
-import Portal from '../components/graphics/portal';
-import {usePrivy} from '@privy-io/react-auth';
-import Head from 'next/head';
+import Portal from "../components/graphics/portal";
+import { usePrivy } from "@privy-io/react-auth";
+import { PrivyClient } from "@privy-io/server-auth";
+import { GetServerSideProps } from "next";
+import Head from "next/head";
+
+export const getServerSideProps: GetServerSideProps = async ({ req }) => {
+  const cookieAuthToken = req.cookies["privy-token"];
+  if (!cookieAuthToken) return { props: {} };
+
+  const PRIVY_APP_ID = process.env.NEXT_PUBLIC_PRIVY_APP_ID;
+  const PRIVY_APP_SECRET = process.env.PRIVY_APP_SECRET;
+  const client = new PrivyClient(PRIVY_APP_ID!, PRIVY_APP_SECRET!);
+
+  // Use this result to pass props to a page for server rendering or to drive redirects!
+  // ref https://nextjs.org/docs/pages/api-reference/functions/get-server-side-props
+  const claims = await client.verifyAuthToken(cookieAuthToken);
```

Comment:
supernit: should we `try`/`catch` this and then route the user depending on if error or success